community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/FeatureToggles.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.internal.dragons;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+public class FeatureToggles
+{
+    public static boolean flag( String name, boolean defaultValue )
+    {
+        return HACK == null ? defaultValue : flag( HACK.getCallingClass(), name, defaultValue );
+    }
+
+    public static boolean flag( Class<?> location, String name, boolean defaultValue )
+    {
+        return booleanProperty( name( location, name ), defaultValue );
+    }
+
+    public static long getLong( String name, long defaultValue )
+    {
+        return HACK == null ? defaultValue : getLong( HACK.getCallingClass(), name, defaultValue );
+    }
+
+    public static long getLong( Class<?> location, String name, long defaultValue )
+    {
+        return Long.getLong( name( location, name ), defaultValue );
+    }
+
+    public static int getInteger( String name, int defaultValue )
+    {
+        return HACK == null ? defaultValue : getInteger( HACK.getCallingClass(), name, defaultValue );
+    }
+
+    public static int getInteger( Class<?> location, String name, int defaultValue )
+    {
+        return Integer.getInteger( name( location, name ), defaultValue );
+    }
+
+    public static <E extends Enum<E>> E flag( String name, E defaultValue )
+    {
+        return HACK == null ? defaultValue : flag( HACK.getCallingClass(), name, defaultValue );
+    }
+
+    public static <E extends Enum<E>> E flag( Class<?> location, String name, E defaultValue )
+    {
+        return enumProperty( defaultValue.getDeclaringClass(), name( location, name ), defaultValue );
+    }
+
+    // <implementation>
+
+    private static String name( Class<?> location, String name )
+    {
+        return location.getCanonicalName() + "." + name;
+    }
+
+    private static boolean booleanProperty( String flag, boolean defaultValue )
+    {
+        return parseBoolean( System.getProperty( flag ), defaultValue );
+    }
+
+    private static boolean parseBoolean( String value, boolean defaultValue )
+    {
+        return defaultValue ? !"false".equalsIgnoreCase( value ) : "true".equalsIgnoreCase( value );
+    }
+
+    private static <E extends Enum<E>> E enumProperty( Class<E> enumClass, String name, E defaultValue )
+    {
+        try
+        {
+            return Enum.valueOf( enumClass, System.getProperty( name, defaultValue.name() ) );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            return defaultValue;
+        }
+    }
+
+    private static final Hack HACK;
```

Comment:
I would much rather you provided some constructive feedback on what a better name would be.